main.py

```diff
@@ -214,11 +213,12 @@ def cost_fn(params, mask=None):
         # get the real gradients as gradients also contain values from dropped gates
         real_gradients = masked_params.apply_mask(gradient)
 
-        print(
-            "Step: {:4d} | Cost: {: .5f} | Gradient Variance: {: .9f}".format(
-                step, current_cost, np.var(real_gradients[0:current_layers])
+        if __debug__:
+            print(
+                "Step: {:4d} | Cost: {: .5f} | Gradient Variance: {: .9f}".format(
+                    step, current_cost, np.var(real_gradients[0:current_layers])
```

Comment:
Consider an f-string:
```suggestion
                f"Step: {step:4d} | Cost: {current_cost: .5f} | Gradient Variance: {np.var(real_gradients[0:current_layers]: .9f}"
```